omim2obo/parsers/omim_txt_parser.py

```diff
@@ -116,14 +116,110 @@ def convert_txt_to_tsv(file_name: str):
     df.to_csv(mim_file_tsv_path, sep='\t', index=False)
 
 
-def get_mim_file(file_name: str, download=False, return_df=False) -> Union[List[str], pd.DataFrame]:
+# TODO: branch into 2 sep funcs if indeed need to update 2 files
+#  - if only 1 file needed, drop file_name param
+# TODO: do we need to update genemap2? is that even being used?
+#  - if so, need to add here, and in 1 place in get_mim_file()
+def update_mim_file_with_protected(
+    file_name: str, inpath: str, outpath: str, protected_path=DISEASE_GENE_PROTECTED_PATH
+):
+    """Update the files downloaded from OMIM to add information we've set in protected-disease-gene.tsv
+
+    Information about this 'protected' file can be found in the docs (README.md).
+
+    mim2gene.txt: As of 2025/03/05, it turned out there were no missing rows in this file. So this code is here now more
+    for futureproofing purposes.
+    """
+    outpath_with_header = outpath.replace('.tsv', '-with-header.tsv')
+    df = pd.read_csv(inpath, comment='#', sep='\t', dtype=str).fillna('')
+    df['is_added_protection'] = False
+    prot_df = pd.read_csv(protected_path, sep='\t').fillna('')
+
+    # TODO: is this needed by morbidmap? if not, move below
+    hgnc_id_symbols: Dict[str, str] = get_hgnc_id_symbol_map()
+
+    if file_name == 'morbidmap.txt':
+        prot_df = prot_df[prot_df['type'] == 'digenic']
```

Comment:
@joeflack4 can you explain why this is limited to entries where the`type` is `digenic` versus using the phenotype mim from the "protected" file as the filter? Any new kinds of "protected" gene associations should not need to be filtered by a `type` value that would need an update to the code vs. only needing an update of content to the "protected" file, ie `protected-disease-gene.tsv`.

Reply:
See: [comment](https://github.com/monarch-initiative/omim/pull/198#discussion_r1986507582)